src/Morphir/TypeScript/NamespaceMerger.elm

```diff
@@ -0,0 +1,107 @@
+module Morphir.TypeScript.NamespaceMerger exposing (mergeNamespaces)
+
+{-| This module contains the TypeScript backend that translates the Morphir IR into TypeScript.
+-}
+
+import Morphir.TypeScript.AST exposing (Privacy(..), TypeDef(..))
+
+
+mergeNamespaces : List TypeDef -> List TypeDef
+mergeNamespaces inputList =
+    let
+        isNamespace : TypeDef -> Bool
+        isNamespace typeDef =
+            case typeDef of
+                Namespace _ ->
+                    True
+
+                _ ->
+                    False
+
+        nonNamespaces : List TypeDef
+        nonNamespaces =
+            inputList |> List.filter (isNamespace >> not)
+
+        originalNamespaces : List TypeDef
+        originalNamespaces =
+            inputList |> List.filter isNamespace
+
+        hasMatchInList : List TypeDef -> TypeDef -> Bool
+        hasMatchInList typeDefList typedef =
+            case typedef of
+                Namespace first ->
+                    typeDefList
+                        |> List.any
+                            (\secondType ->
+                                case secondType of
+                                    Namespace second ->
+                                        second.name == first.name
+
+                                    _ ->
+                                        False
+                            )
+
+                _ ->
+                    False
+
+        emptyListOfNamespaces : List TypeDef
+        emptyListOfNamespaces =
+            []
+
+        insertNameSpaceIntoList : TypeDef -> List TypeDef -> List TypeDef
+        insertNameSpaceIntoList insertDef listTypeDef =
```

Comment:
inconsistent capitalisation: elsewhere we use Namespace

Reply:
fixed